src/Application/Game.cpp

```diff
@@ -2502,17 +2502,16 @@ void Game::EventLoop() {
         }
     }
 
-    pMessageQueue_50CBD0 = pMessageQueue_50C9E8;
-    pMessageQueue_50C9E8->Clear();
+    std::swap(pCurrentFrameMessageQueue, pNextFrameMessageQueue);
```

Comment:
Does this need any error checking to make sure queue is empty?

Reply:
Probably should add `assert(pNextFrameMessageQueue->Empty());`

Reply:
I could but I think it's redundant because loop that process queue (the one above this code) works until current frame queue is not drained.
So after loop is exited we will have current frame queue is empty and next frame queue possibly not empty (which is intended).

Reply:
Yeah makes sense.

Reply:
OK I've looked at the code, and this loop is so big that you need to be a bit insane to be able to follow what's going on. Let's add an assert, just to feel safe.